### ISSUE TITLE
Add support for Pipelines in repos with configurable input and output

### ISSUE THREAD

**Issue by widhalmt** [NETWAYS/ansible-role-logstash#94]
# TL;DR #

Why not add `.gitignore` to every pipeline repository making `input.conf` and `output.conf` ignored. So every user could build their own. And this role could optionally just create these files.

# The gory details # 

We now have the following:

* Some default configuration for pipelines we offer for extreme quick setup if you don't want to dig into Logstash configuration at allbefore starting
* Checking out and including of complete pipelines via git
* "Connector pipelines" ( https://github.com/NETWAYS/ansible-role-logstash/blob/master/docs/connector-pipelines.md ) that can't have filters but automatically set `input`s, `output`s and `if`
* Pipeline directories (with inclusion into `pipelines.yml`) where it's totally up to the user to fill them with code

What now if we find yet another way of providing easy to use pipelines. "Connector Pipelines" were invented because we couldn't find a way of making `key` in `input` and `output` configuration easily configurable for repositories in git. So we built hardcoded `key` settings and invented "connector pipelines" to make up for hardcoded keys.

But why not just provide pipelines without "input" and "output" configuration? The problem we had that it would mean to break compatibility to upstream repositories and would make updating a mess. The solution could be to simpy provide a `.gitignore` file with the repositories that includes e.g. `input.conf` and `output.conf`. Then a user could set up their own input and output configuration and still use our (or their) git managed pipeline configuration.

This new version of pipelines could be made out of existing pipelines without breaking downwards compatibility. When we add the `.gitignore` and a user already has an older version `git` should leave the version they already have untouched (give that a test before commiting). New users would have to place their own versions of `input` and `output`.

Why I even mention this here: We can improve the feature that fetches pipelines from git repositories in a way that you could optionally add names for keys for `input` and `output`. The role could then just place `redis` configuration in the same directory. This way we don't lose anything in existing setups, lose no functionality for new setups (save for adding new subkeys to set) and could make setups way easier and less ressource hungry. On the other hand we already have connector pipelines that would help with more complex setups or pipelines that don't follow this potential new schema.
